tools/doc/allhtml.mjs

```diff
@@ -31,13 +31,35 @@ for (const link of toc.match(/<a.*?>/g)) {
   // Split the doc.
   const match = /(<\/ul>\s*)?<\/\w+>\s*<\w+ id="apicontent">/.exec(data);
 
+  // Get module name
+  const moduleName = href.split('.')[0];
+
   contents += data.slice(0, match.index)
-    .replace(/[\s\S]*?id="toc"[^>]*>\s*<\w+>.*?<\/\w+>\s*(<ul>\s*)?/, '');
+    .replace(/[\s\S]*?id="toc"[^>]*>\s*<\w+>.*?<\/\w+>\s*(<ul>\s*)?/, '')
+    // Prefix TOC links with current module name
+    .replace(/<a href="#([^"]+)"/g, (match, anchor) => {
+      return `<a href="#${moduleName}_${anchor}"`;
+    });
 
   apicontent += '<section>' + data.slice(match.index + match[0].length)
     .replace(/<!-- API END -->[\s\S]*/, '</section>')
+    // Prefix all in-page id's with module name (mostly special cases)
+    .replace(/id="([^"]+)"/g, (match, anchor) => {
+      return `id="${moduleName}_${anchor}"`;
+    })
```

Comment:
The issue with that is that we have some "hard-coded" `id`s in our MarkDown docs, not sure how we should deal with them

Reply:
I think all the hardcoded ID's still work in the generated HTML (almost all of them are identical to the new heading anchors, differing only in case). Do you have an example of one that's breaking?

Reply:
E.G.: https://nodejs.org/docs/latest/api/all.html#inspector_security

When I compile your version, http://127.0.0.1:8000/all.html#inspector_security doesn't target the correct section – instead, it has become http://127.0.0.1:8000/all.html#cli_inspector_security

Reply:
Thank you and agreed. I've removed the general prefixing of all hard-coded id, which should preserve all incoming links to those anchors.

The deprecation links already have [special processing during HTML conversion](https://github.com/nodejs/node/blob/master/tools/doc/html.mjs#L404), so I'm now [ignoring those](https://github.com/nodejs/node/pull/39304/commits/6260f08d42d4c4b628e3b439adbf0b232452b80b), which should fix all the `DEP` anchors.

I've fixed a few more special cases for `all.html` in 28d371c3f28dcc1ccd769b755486a18b203356da. Hopefully this covers your concerns!